Test failures in librsynctest.py with librsync 2.2.x and 2.3.x

I'm trying to build run rdiff-backup 2.0.0's test suite and got the following errors:

```
======================================================================
FAIL: testSigFile (__main__.LibrsyncTest)
Make sure SigFile generates same data as rdiff, blocksize 512
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testing/librsynctest.py", line 62, in testSigFile
    self.sig_file_test_helper(512, 5)
  File "testing/librsynctest.py", line 51, in sig_file_test_helper
    assert rdiff_sig == librsync_sig, \
AssertionError: (1668, 1668)

======================================================================
FAIL: testSigFile2 (__main__.LibrsyncTest)
Test SigFile like above, but try various blocksize
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testing/librsynctest.py", line 66, in testSigFile2
    self.sig_file_test_helper(2048, 1, 60000)
  File "testing/librsynctest.py", line 51, in sig_file_test_helper
    assert rdiff_sig == librsync_sig, \
AssertionError: (372, 372)

----------------------------------------------------------------------
```

It turns out the length is the same but the content is quite different:

```
('b\'rs\\x01F\\x00\\x00\\x02\\x00\\x00\\x00\\x00\\x08\\xee\\x85w\\xd7\\xd6\\x9d\\xf8Y\\xce\\x9d\\x7f\\xfb\\xcb\\xb8R\\x10b\\xf8\\x07b\\xa6\\x0f\\xbc=\\xf2J\\xa6\\xc9\\x14~\\xcd&m\\xd5M\\x89OXonw2wR\\x03\\x91\\xa9\\xddS\\xf3\\x1eR\\xefgY\\xaa\\xe8\\x0b~@Ibt;\\x88\\xd5\\xa7\\x01\\xe3;\\x91u\\xca@\\x07VW\\xb2\\x92\\xfc\\xab/\\xfe\\x9cQ\\xebM\\x00<\\x07)\\xb2Ply4\\x0f\\xc6\\xc2\\xc9\\xcbqD$\\xa7\\x0f8\\x86i\\\\\\xdd\\xe4\\xcf\\xceY\\xd8]\\xc0\\xa5;\\xcf\\xb9\\xe9\\xe9\\xedH\\x80\\x07\\x80\\x1fq\\xb7\\x97\\xc1^( _\\x08\\x91Y\\x07\\x80\\x9c\\x93?_&\\x87\\x12\\x8a\\x8f\\xaf\\x97dh\\xb8\\xea\\xf4\\x8c\\x12;t)\\xd2\\xc9F\\x9b]YU\\xceH\\xe2\\xbf\\x1d\\xfbY\\xec6\\xeb\\x05\\x97\\x94\\xfaQ\\x1f\\x12\\x1c\\x98:\\xd5\\xf7\\xcb,\\xe8EL\\xb3\\x1d\\x95H\\xea\\xf3\\x9f+\\xcd\\xd7{\\xa61\\xa8\\x81^\\xaf\\\'lME\\xb9\\xeb\\x99\\x18\\xfb^%G\\xed\\xc7\\x9e\\x88~\\xbd\\xacAY\\x9b\\xf1\\xb9\\xc9^\\xbe\\xc1\\x11\\xdf\\x86\\x187\\xf7\\xc1\\xe9\\x0e\\xaf\\xbf\\x89\\x16:\\xb3G\\xdbW\\xe0\\xb2\\x89\\x96 +\\x89\\xfcmn[\\xcf\\xe1\\xae\\x93T\\x95\\x17\\x82}\\xfdY=\\x9a\\x16\\x1e\\xe8\\xf0d\\x18\\x85C\\x99)\\xab\\x85"I\\xe2=\\xa3\\xbdg\\xe4p\\xa7Y~\\xb0^\\xc2r\\x0b\\x0c\\xa6)\\x1c\\x18\\xf3<U\\xb8(\\x1e\\x81\\x894b>\\x99\\x0e\\xf0\\xe7w\\x92\\x8a6Y;.2\\x8e"\\x0e\\\'\\x0f\\xc0~\\x0fz,\\xec\\xb5\\x05\\xfe\\xf1\\xea\\xb01J)\\x1a\\x0f\\x8f\\t\\xf0\\xb4<}o\\x96\\xe2\\x13\\xd8\\xb7\\xe7W\\xbd\\x19\\x86>\\x074C\\xb8\\xf6\\xbd\\x9a(j\\xbd\\xe5\\xa4\\xf1\\xd6K\\xe0\\xbe\\xda^\\x83\\xcep\\xef\\xdc\\xd2$\\xd6\\x80H\\xbbp\\x99\\xaf-\\xe4\\xf4S\\x02\\xdc\\x04\\xa8b6=[l\\xa3\\t\\x85\\x93g+ID\\xde\\xc7\\x11@\\r\\x9c\\xe9\\x9c\\xa6\\x8e7\\xb7\\x92\\xb8\\x86J\\x8f\\x95\\xf5\\x80Ddt\\xa4(O\\xae\\x91\\xf1M6\\xcaW\\x89\\xb9\\xbfjhE\\x9d\\xfb\\xda\\xab?\\xb61\\xc4\\x86^\\xe9lU\\x92\\x9c\\x9e\\xf9\\x84\\x0f\\xd4\\x02\\xb8W\\xc2\\xf0\\xc0\\xf3\\xf3\\xf5\\x03$:\\xaf\\xde\\xcf\\x14\\xeby\\xcc\\xc8\\x9f\\x13w\\xf5U\\xaf\\x99\\xbc\\x8a>\\xe6\\xf6Y\\xf1\\x19\\x80\\x05\\x843\\xd9\\x9c\\x1e\\x06b\\xd7\\x7f\\xa9\\x1bX\\xb2\\x85\\xe3AP\\x0f\\xc3u\\xe4\\xc7\\xdaD[A2/j\\xc3>\\xb9\\xd1\\xa6\\xe0\\xd5\\xbbI+\\xc5\\xa0\\x9c~\\xd1\\xfc\\xdc\\xc4\\xab\\x1f*\\xa9Y\\x87G-\\x9f\\x919\\x01j\\xbfY3\\xeed.u\\xcc\\xfc^\\xfe\\xf3\\x93B0\\x98d\\xea\\xfd*\\xf6\\x87f>1\\x03\\t\\xd5\\x9c"\\x8f\\x12\\xe72\\xd6s\\x17\\x9clS#\\x97z!j2Tb"\\x12\\x19\\xced\\xb0aq\\x91\\xf1\\xd9I\\x07%\\xa0\\xef\\x05\\xe8\\xc9\\xab\\x01\\xf5\\xe5\\x10\\xaf\\x91\\xa2\\x8b\\xa6S\\xe1p1m\\xd5/\\x8b\\xc9\\xdd\\xc8\\x92\\x8bL\\trS.\\xef\\x04\\x82\\x07\\xcd\\xe4?\\xac\\xf1\\x0e\\xad+:\\xc0k\\xc6H3Ll\\xa5\\xdcK\\xefNL\\x1a}\\xf9\\x1b~\\x84\\xed\\xb2\\xb7jc\\xf7S\\x8fdL\\x0e\\xdb\\xf4k\\x05\\xfb\\x00\\xa4\\x1eQ\\xb6r\\x86\\xd9\\x96~\\xb5\\xfe\\x02\\x94w\\x05\\x0f\\x04.\\xd1w#`/\\xc1@\\xe3si<qT\\x0c\\x04Y\\xd7\\xfa\\xe8\\xfb\\xd4{i\\xa3\\x1f&\\xa1\\x82aUKD\\xcb\\xb4\\xad\\xc2\\xc5\\xc3\\xe4\\xfa\\xe5\\xd3\\xd4\\t\\x98\\xcd\\xb5\\xd4,\\x0f#\\xa4\\x9b\\xc6\\\'\\xb8YU\\x81;/\\x0c\\x7f.\\xe8\\xf9\\x96p\\xe9*15E\\xe9\\\\e\\xf7C\\xe8.+\\x84\\x1c:P\\xc8kU\\x13\\xd1\\x9a\\xe5\\xa4\\x9bc\\xb0\\xf3\\xc64\\xc5\\xe8\\x9b\\x8e\\xc4\\x99\\xac@\\xb0\\xbe\\xb0V\\xd1|\\xa1\\x8e}?\\xafY\\x15\\xaa\\x8d\\xf3\\x8e\\x88\\x89i\\xde\\xfb8\\x82\\x8c\\xca\\xa9\\xf6\\x0f\\x1cW\\xc0M\\x9c4\\xf7x\\xc9G=p\\x87\\x92i\\xb9\\xb9\\x9b\\xb1A\\xc4\\xfbwy\\xe1\\x90}lH1BDY\\x90\\xf1\\xe8\\xc6\\xc6\\x87\\x8dm\\x1d\\x93\\x86I\\xe7\\xe4\\xa3\\xd6\\rc\\x93/=\\x05\\x12\\xf0T\\xeb\\x9e\\x00b\\xc32o\\xb6\\x133\\xb0\\xd3M\\x95X\\xf8\\x93w\\x01/\\x85\\xe8\\x932\\\\\\xc6Y\\xd5\\xa5\\xd9\\xe8"\\xfc\\x08\\xd38\\x11\\x8c\\xbe\\xfdt]?i\\xf74\\x8c\\xef\\x82:\\xeb#\\xa5\\x04`\\x92*%\\x8e\\xe9\\x0eo\\xa5\\xc0BV{\\x9e&\\x9bdhBJ\\xea\\xd2C\\xc5\\xb6\\x82\\x154H\\x16)\\xccT\\xee\\xd1\\xac\\x8c\\x88\\xba\\x98\\xff\\x8d\\xc0X\\xf9\\x93\\xc0\\xb2\\x98a\\xb1~\\xa2\\xe5i\\x94X#\\x1d9\\x8e\\x07\\xf4\\xdb\\x95:G\\x02CBb\\x1d\\xa0\\x07\\xed\\xcd\\x953\\x8f\\x180\\x94\\x11\\x84CN55O&@\\xa7*\\xa4\\r\\xa8\\x07\\xc4q$\\xaa\\xb4X\\x19R\\x12s\\x08\\xd4[\\xec~\\xa4\\xa3\\xf4]\\xea\\x07\\x8b\\xa6\\t@\\xc4\\xeb\\xaeB\\x88`t\\xfa!\\x06:\\x00\\xd5\\xac\\xce\\xfbo\\x81\\x12\\xdcY\\xec\\nq\\xdd\\xe5~\\x8e^\\xf4\\xba\\x85uLjH\\xc9\\x84\\xa0\\x10\\xf1O \\xfdMP\\\'~l*\\xc4\\xb7\\xaf\\x855\\xb5#\\x93\\x852\\xe9"\\xa2\\xc4"\\xc8^\\xff\\xce\\x18\\xd8\\xfe\\x9d\\x83[\\xb1\\x04_\\xf9\\t\\xad2?{(\\x99\\x02\\x90\\x1b\\xec\\xe9\\x84\\x8f\\xdb\\xef\\xe7\\xd1z4K\\x1e\\xf1}Z\\xc9#6\\xffi\\x02\\x8e\\xbe\\xe2\\xa0\\x81\\\\Y\\x8a\\xc9@\\xa79\\xd7\\x06!\\xd9\\x15\\xfc\\xa6\\xeb+N\\xdc|\\xca\\x9c\\xc7\\xaa\\x1eG\\xe1\\x90D\\nU\\x86JC\\x84\\x1a\\xb9\\xbe?\\x01MD\\xa3ng<\\xdf\\xe4\\xbb\\xae\\xdc.\\xc6\\x97\\xa0"\\xec<(\\xd3)l\\xc8\\x88\\xd6\\x19\\x88\\x1e4\\x0c\\xee\\xc7\\xb8\\x91q\\xbf\\x94P\\xd2\\x03\\xb8\\x11\\xb5\\r\\xd0\\xb1\\x1d-U\\x13\\x92F\\t\\x18\\xef\\x82\\xa8\\xa4$\\xb3\\xe5\\xb3"\\x15\\x11\\x88a^\\x1d&\\xf4QgTb|\\x1e\\xc6c\\xb4[\\x85\\xb1-r\\x11-\\x19\\xd3\\x1a\\x04\\x06&\\xc9\\xaa\\xc6\\xda(Y\\xdd\\x92\\xca-\\x11\\xc1)\\xafL\\xc3\\xb7#\\xff\\x06\\xd0\\xf1\\x00\\xd8t\\xe9\\xb4\\xc8\\xe5\\xf7\\xe3\\x9fU\\tY\\xceP\\xa8\\x88\\xed\\xa6\\xf9\\x83\\xc22\\xc6\\xf1\\x84F(\\x87\\xe4g(\\xb7XW\\x91\\xc6g\\xd8\\xbf\\x8ec\\x93\\xb0\\x0e\\xfav|\\xd8dc\\x12(\\xcb\\xdf\\xcf\\x9e8\\x1e\\x82U\\xde\\x9a\\xa5\\xf8kx$\\xd8K\\xb8\\xc3\\xe5\\xe5G1\\x89\\xb9\\xce0\\xd4;\\xdeJ\\xcd3\\x1dx\\xb1\\xb1Q\\xe5\\x91T\\xafy\\xbb\\xc0*\\xd7]\\xc4R\\x9d->L\\xd1\\xe5+\\x90\\xdf\\xfdm\\xce\\x8a\\xa6\\t,A\\x8d\\x187\\xf9q\\xa2\\x87\\xac\\xe5Q<.8\\xf1N\\x9f\\x15;&\\xc6#\\xb9\\x05\\x9a\\xca\\x90L\\xd3{\\xa6\\x91*\\x10\\x01\\xbfL\\x16Y\\x15\\xd4\\xed\\x02n\\xbePb~\\xc5Z\\xcc\\x8e\\xaf\\x11\\x0cZ\\xca^\\x82\\x0b\\xfb7\\x19\\xa1\\x92\\xcf+\\xf3\\x84`&c\\xc0\\x969\\xb5ee\\x86n<\\xbc\\xd1\\xa0\\xb8v\\xd3\\xd1\\x81\\xb9\\xd0\\xfe_\\x12@j\\xd6/\\x9e\\xe9q\\x8a\\xdf#\\x04\\x05\\xa0\\x0fr\\x08\\xbdYX~\\xd9\\xe8u\\x0c\\x1a\\xd3\'', 
'b\'rs\\x016\\x00\\x00\\x02\\x00\\x00\\x00\\x00\\x08\\xd1::\\x16\\xd6\\x9d\\xf8Y\\xce\\x9d\\x7f\\xfb\\x81\\xb9E\\xa7b\\xf8\\x07b\\xa6\\x0f\\xbc=\\xf6\\x14Bx\\x14~\\xcd&m\\xd5M\\x89*\\xb6>\\xa9w2wR\\x03\\x91\\xa9\\xdd\\xdeIHa\\xefgY\\xaa\\xe8\\x0b~@\\xccS:v\\x88\\xd5\\xa7\\x01\\xe3;\\x91uPH:\\xc9W\\xb2\\x92\\xfc\\xab/\\xfe\\x9c;\\xc40\\x9b<\\x07)\\xb2Ply44nB\\x80\\xcbqD$\\xa7\\x0f8\\x86\\x17z>\\x07\\xcf\\xceY\\xd8]\\xc0\\xa5;\\xb4\\x89<4\\xedH\\x80\\x07\\x80\\x1fq\\xb7\\x94I=\\xff _\\x08\\x91Y\\x07\\x80\\x9c\\xa2:?!\\x87\\x12\\x8a\\x8f\\xaf\\x97dh\\xcdB;?\\x12;t)\\xd2\\xc9F\\x9b#\\xc1;\\xddH\\xe2\\xbf\\x1d\\xfbY\\xec6\\xfbz0\\x07\\xfaQ\\x1f\\x12\\x1c\\x98:\\xd5\\x01\\xf9C\\x7fEL\\xb3\\x1d\\x95H\\xea\\xf3\\xd3:B\\xe6{\\xa61\\xa8\\x81^\\xaf\\\'553\\xc4\\xeb\\x99\\x18\\xfb^%G\\xedl-=\\x8d\\xbd\\xacAY\\x9b\\xf1\\xb9\\xc9E\\x848\\x00\\xdf\\x86\\x187\\xf7\\xc1\\xe9\\x0e\\x046>\\xa1:\\xb3G\\xdbW\\xe0\\xb2\\x89\\xc0\\x06A \\xfcmn[\\xcf\\xe1\\xae\\x93\\x87m:\\x81}\\xfdY=\\x9a\\x16\\x1e\\xe8\\xdc\\xe8;\\x94C\\x99)\\xab\\x85"I\\xe2\\\'V8\\xb6\\xe4p\\xa7Y~\\xb0^\\xc2\\xed^C\\xc1)\\x1c\\x18\\xf3<U\\xb8(\\xae\\xc94kb>\\x99\\x0e\\xf0\\xe7w\\x926\\r>\\xfe.2\\x8e"\\x0e\\\'\\x0f\\xc0h\\xafD{\\xec\\xb5\\x05\\xfe\\xf1\\xea\\xb01x\\xfc<\\xe6\\x8f\\t\\xf0\\xb4<}o\\x96\\xb1\\xb5>\\xaa\\xe7W\\xbd\\x19\\x86>\\x074\\xe5\\xc2?D\\x9a(j\\xbd\\xe5\\xa4\\xf1\\xd6\\xbc\\xf6F5^\\x83\\xcep\\xef\\xdc\\xd2$\\xae\\xa7<6p\\x99\\xaf-\\xe4\\xf4S\\x02*\\x85>\\x916=[l\\xa3\\t\\x85\\x93\\xe31?{\\xde\\xc7\\x11@\\r\\x9c\\xe9\\x9c\\xa7)A\\n\\x92\\xb8\\x86J\\x8f\\x95\\xf5\\x80\\x8d\\xedF\\x0b(O\\xae\\x91\\xf1M6\\xca\\xd8\\xac<\\xb6jhE\\x9d\\xfb\\xda\\xab?8\\xee?q^\\xe9lU\\x92\\x9c\\x9e\\xf9\\xc7\\xa1=\\xb1\\xb8W\\xc2\\xf0\\xc0\\xf3\\xf3\\xf5\\xd5\\xe7?\\x9a\\xde\\xcf\\x14\\xeby\\xcc\\xc8\\x9f0\\xae2\\\\\\xaf\\x99\\xbc\\x8a>\\xe6\\xf6Y\\x89)D\\x90\\x843\\xd9\\x9c\\x1e\\x06b\\xd7\\xf7\\x88>\\x03\\xb2\\x85\\xe3AP\\x0f\\xc3u\\x94UC\\xeb[A2/j\\xc3>\\xb9\\x16\\x0b;\\xc8\\xbbI+\\xc5\\xa0\\x9c~\\xd1#\\x083\\xe2\\x1f*\\xa9Y\\x87G-\\x9f\\t\\x99;\\x19\\xbfY3\\xeed.u\\xcc\\xfe:?2\\x93B0\\x98d\\xea\\xfd*\\xca?=\\xc51\\x03\\t\\xd5\\x9c"\\x8f\\x12\\x8a"8\\x02\\x17\\x9clS#\\x97z!\\xe3%0\\x01"\\x12\\x19\\xced\\xb0aq\\xe9\\xe39\\\\\\x07%\\xa0\\xef\\x05\\xe8\\xc9\\xab\\x8f\\xe2>\\xe3\\xaf\\x91\\xa2\\x8b\\xa6S\\xe1p\\x1a\\x889V\\x8b\\xc9\\xdd\\xc8\\x92\\x8bL\\t\\xa9\\xccLf\\x04\\x82\\x07\\xcd\\xe4?\\xac\\xf1Dj?\\x85\\xc0k\\xc6H3Ll\\xa5:\\xc3KEL\\x1a}\\xf9\\x1b~\\x84\\xedv\\xbdM\\x96\\xf7S\\x8fdL\\x0e\\xdb\\xf4e\\xe04\\xeb\\xa4\\x1eQ\\xb6r\\x86\\xd9\\x96\\xc8\\x99;\\x81\\x94w\\x05\\x0f\\x04.\\xd1wrCH$@\\xe3si<qT\\x0c \\x856i\\xe8\\xfb\\xd4{i\\xa3\\x1f&\\x04\\xf6:\\xfcKD\\xcb\\xb4\\xad\\xc2\\xc5\\xc3\\x98k3\\x8e\\xd4\\t\\x98\\xcd\\xb5\\xd4,\\x0fLdH\\xe9\\\'\\xb8YU\\x81;/\\x0c\\xd2\\x11;\\xc4\\x96p\\xe9*15E\\xe9\\\'$<\\n\\xe8.+\\x84\\x1c:P\\xc8\\x14\\xd5D\\xec\\x9a\\xe5\\xa4\\x9bc\\xb0\\xf3\\xc62U3n\\x8e\\xc4\\x99\\xac@\\xb0\\xbe\\xb0\\xeaK<\\xf4\\x8e}?\\xafY\\x15\\xaa\\x8d\\xfcS;|i\\xde\\xfb8\\x82\\x8c\\xca\\xa9\\xfa\\xab1r\\xc0M\\x9c4\\xf7x\\xc9G\\xe7e=\\x01i\\xb9\\xb9\\x9b\\xb1A\\xc4\\xfb\\x0c\\xda<\\xd3}lH1BDY\\x90g\\x00<\\xb9\\x87\\x8dm\\x1d\\x93\\x86I\\xe7\\x19w<\\xa0c\\x93/=\\x05\\x12\\xf0T\\xadsG\\xa9\\xc32o\\xb6\\x133\\xb0\\xd3\\x86\\xb7;w\\x93w\\x01/\\x85\\xe8\\x932\\xa4\\x95>\\x10\\xa5\\xd9\\xe8"\\xfc\\x08\\xd38\\xd6b<Dt]?i\\xf74\\x8c\\xef\\xe6\\xd9E\\xe6\\xa5\\x04`\\x92*%\\x8e\\xe9qq8\\\'BV{\\x9e&\\x9bdh\\x9c\\xc16\\xa1C\\xc5\\xb6\\x82\\x154H\\x16\\xfa>?I\\xd1\\xac\\x8c\\x88\\xba\\x98\\xff\\x8d\\xc5\\xa0E\\x1a\\xc0\\xb2\\x98a\\xb1~\\xa2\\xe5"\\xea=r\\x1d9\\x8e\\x07\\xf4\\xdb\\x95:hW<\\x19b\\x1d\\xa0\\x07\\xed\\xcd\\x953n~4W\\x11\\x84CN55O&\\\'\\xa5>[\\r\\xa8\\x07\\xc4q$\\xaa\\xb4`\\x8dE\\xf1s\\x08\\xd4[\\xec~\\xa4\\xa3\\xe3MDJ\\x8b\\xa6\\t@\\xc4\\xeb\\xaeBw\\x0cD\\xfd!\\x06:\\x00\\xd5\\xac\\xce\\xfbM\\xce>OY\\xec\\nq\\xdd\\xe5~\\x8e\\x87\\xf970uLjH\\xc9\\x84\\xa0\\x10\\xe9}9hMP\\\'~l*\\xc4\\xb73\\x02F|#\\x93\\x852\\xe9"\\xa2\\xc4\\xeadD\\xc6\\xce\\x18\\xd8\\xfe\\x9d\\x83[\\xb1L\\xc67\\x80\\xad2?{(\\x99\\x02\\x90\\xdc\\xcd2\\x0b\\x8f\\xdb\\xef\\xe7\\xd1z4KP:?%\\xc9#6\\xffi\\x02\\x8e\\xbe[N>\\xdfY\\x8a\\xc9@\\xa79\\xd7\\x06\\\\\\xbdB\\xf3\\xa6\\xeb+N\\xdc|\\xca\\x9c\\x94\\x1f@\\xf2\\xe1\\x90D\\nU\\x86JC\\x9er>\\xa9?\\x01MD\\xa3ng<\\x92\\x06GI\\xdc.\\xc6\\x97\\xa0"\\xec<|\\xf1:\\x13\\xc8\\x88\\xd6\\x19\\x88\\x1e4\\x0c(\\xb04Pq\\xbf\\x94P\\xd2\\x03\\xb8\\x11\\x17\\x14>\\xb0\\x1d-U\\x13\\x92F\\t\\x18\\xa7\\xccCo$\\xb3\\xe5\\xb3"\\x15\\x11\\x88\\xd6\\xcd2\\xb5\\xf4QgTb|\\x1e\\xc6\\xf5\\x99<\\x10\\xb1-r\\x11-\\x19\\xd3\\x1a\\xf1\\x9c?8\\xaa\\xc6\\xda(Y\\xdd\\x92\\xca1#B\\\\\\xafL\\xc3\\xb7#\\xff\\x06\\xd0\\xa8\\xa34\\xa3\\xe9\\xb4\\xc8\\xe5\\xf7\\xe3\\x9fU\\xfa!?W\\xa8\\x88\\xed\\xa6\\xf9\\x83\\xc22\\xd8\\x8fFM(\\x87\\xe4g(\\xb7XW\\xd0F:\\xeb\\xbf\\x8ec\\x93\\xb0\\x0e\\xfav*\\xc67\\x92\\x12(\\xcb\\xdf\\xcf\\x9e8\\x1e\\xae47\\xed\\xa5\\xf8kx$\\xd8K\\xb8A\\xb3AB1\\x89\\xb9\\xce0\\xd4;\\xde*\\xe28\\xe4x\\xb1\\xb1Q\\xe5\\x91T\\xafW\\x89G\\x99\\xd7]\\xc4R\\x9d->Lmz4\\xc3\\xdf\\xfdm\\xce\\x8a\\xa6\\t,+2>\\x96\\xf9q\\xa2\\x87\\xac\\xe5Q<\\x19\\xe3Ae\\x9f\\x15;&\\xc6#\\xb9\\x05V\\x94;\\x87\\xd3{\\xa6\\x91*\\x10\\x01\\xbf?\\xe0<\\x14\\xd4\\xed\\x02n\\xbePb~\\x8f\\x94>\\xc1\\xaf\\x11\\x0cZ\\xca^\\x82\\x0bXXA\\xd0\\x92\\xcf+\\xf3\\x84`&cF\\xf15\\xc0ee\\x86n<\\xbc\\xd1\\xa0\\xec(: \\x81\\xb9\\xd0\\xfe_\\x12@j\\x00\\rA\\x94q\\x8a\\xdf#\\x04\\x05\\xa0\\x0f\\xaa\\xda\\xe7]X~\\xd9\\xe8u\\x0c\\x1a\\xd3\'')
```

```
'b\'rs\\x01F\\x00\\x00\\x08\\x00\\x00\\x00\\x00\\x08\\x1e\\x07\\x1e\\x0fBY\\x07\\xe5z\\x08O\\x82\\xee\\x82\\\\}~\\xc3dQ6\\xb7\\x94\\xf9\\xdcE\\x01\\xc1\\x82g\\xdb\\xfb\\xf1w\\x816bz=\\xc9\\xd8\\x92%U\\x95\\xee\\xa8\\xbd\\x9b\\xb0\\x12\\xd9u\\x1f,HYe\\x9f\\x1a\\xba\\xf7\\xe0\\x86[\\xe64u\\xbb\\x8f~=\\x16\\n*\\x0b\\xdbJ\\xa2O\\xec\\x04E\\xc2\\xe4\\xa1\\xf09\\xf0\\xce\\xf6\\r\\x90\\xcb\\xca\\x13t \\x1e)\\x07\\xee\\x08\\xcc{\\xe5\\x9a^\\x83\\x82\\xc6\\x8f\\x9b\\x9a\\x11N\\x1c~\\xddq\\xc1\\xf1o\\x86\\xb1\\xca\\x1e\\x00\\xae\\xdd\\xb1pJ/\\xc3\\xc2[\\xbdJ\\xb0\\xa8\\xf2\\xb4\\xef2r\\xae\\xf9\\r?7\\xd5\\xa2\\x07\\xbb0\\xa4\\xa0\\xc3\\xa2\\x0f\\xe9\\x0e\\xf9\\xb5F8\\x99\\x98\\x1f\\\'\\xaa\\x84\\xd1\\x02\\xa0KJ?t#Cl\\x82\\x06\\xe1E\\xc9o\\xeb\\xb16\\x99^\\xfd7\\xfbM\\x97v\\x1c\\xfe\\xcf\\x8a!O\\xb3[\\xa1\\x8f2s\\x7f\\xb9ttl\\x8b\\xde8\\x9f\\xbb\\x10n\\x83\\x1cM\\xf5N\\x85\\xa5\\xb4m\\xb6\\xd2k\\xef`Jq\\xa1\\x0c\\xf4\\xa6*\\xd4\\xe1\\xfc\\xd3e\\xb0[l.\\xf0\\xad\\xde\\x9bL\\x10\\xfcHWh\\xf7\\x1d\\x17\\x81\\xa7zQU\\x0c\\xf7\\x8f\\xbbR\\xf5\\x8a\\xadRy\\xd0\\x8a\\xb5\\xc7Z||\\xc1`r\\x90D\\x82S\\x0c\\xa1\\xef\\xf8-E6\\xe08\\xe1-\\xef\\xb8!Ta\\xdd\\xca\\xb7\\x8d\\xb6\\xe6\\x8c\\x9dV\\xb55\\x0c|.\\xdb\\x05\\xa1\\xfa\\x9f $\\xdb\\x8f"/\\x0cZx\\xacs^_\\xf4L\\xe6\\x18D\\x00\\x93\\xfa\\xdc\\xc3e3\\xff\\xbd\\xbfz\'', 
'b\'rs\\x016\\x00\\x00\\x08\\x00\\x00\\x00\\x00\\x08{l\\xfbfBY\\x07\\xe5z\\x08O\\x82\\xb8\\xa3\\xf8\\x10~\\xc3dQ6\\xb7\\x94\\xf9\\x17S\\xf4\\xf4\\x82g\\xdb\\xfb\\xf1w\\x816f\\xec\\t\\x98\\xd8\\x92%U\\x95\\xee\\xa8\\xbd\\x95\\xdd\\xf4du\\x1f,HYe\\x9f\\x1a\\xbb\\x17\\x0b\\xcd[\\xe64u\\xbb\\x8f~=!7\\x02F\\xdbJ\\xa2O\\xec\\x04E\\xc2\\xceJ\\xf0\\x00\\xf0\\xce\\xf6\\r\\x90\\xcb\\xca\\x13\\xc0\\x1a\\x020\\x07\\xee\\x08\\xcc{\\xe5\\x9a^\\x04S\\xdd\\n\\x9b\\x9a\\x11N\\x1c~\\xddq\\xc5C\\xfd\\r\\xb1\\xca\\x1e\\x00\\xae\\xdd\\xb1p\\r\\xb9\\xf7\\x81[\\xbdJ\\xb0\\xa8\\xf2\\xb4\\xef^O\\xef\\x1c\\r?7\\xd5\\xa2\\x07\\xbb0\\xbc6\\xf6\\xdd\\x0f\\xe9\\x0e\\xf9\\xb5F8\\x99\\xe3\\xcf\\xf6\\x81\\x84\\xd1\\x02\\xa0KJ?t\\xf3\\x8f\\xed\\xc9\\x06\\xe1E\\xc9o\\xeb\\xb16\\xad\\xb8\\xfa\\x9e\\xfbM\\x97v\\x1c\\xfe\\xcf\\x8a\\x115\\xf1\\x0e\\xa1\\x8f2s\\x7f\\xb9tt\\xcd\\x86\\xea+\\x9f\\xbb\\x10n\\x83\\x1cM\\xf5y\\x0b\\xde\\xd3m\\xb6\\xd2k\\xef`Jqf\\x81\\xff\\x95*\\xd4\\xe1\\xfc\\xd3e\\xb0[\\xfdg\\xee\\x80\\xde\\x9bL\\x10\\xfcHWh\\xe8\\x8b\\xfd\\xc4\\xa7zQU\\x0c\\xf7\\x8f\\xbbF\\xf4\\xf1\\xbcRy\\xd0\\x8a\\xb5\\xc7Z|q\\xa5\\xe8\\x91\\x90D\\x82S\\x0c\\xa1\\xef\\xf8.y\\xe478\\xe1-\\xef\\xb8!Ta\\xfah\\xfe\\x1c\\xb6\\xe6\\x8c\\x9dV\\xb55\\x0c\\x04$\\xfd4\\xa1\\xfa\\x9f $\\xdb\\x8f"\\xfe\\x83\\xf4\\xe7\\xacs^_\\xf4L\\xe6\\x18\\xda:|\\x05\\xdc\\xc3e3\\xff\\xbd\\xbfz\''
```

I tried to run the tests locally with `tox -e py38` and got exactly the same error.

```
py38 run-test: commands[2] | python testing/librsynctest.py
/home/felix/projects/rdiff-backup/.tox/py38/lib/python3.8/site-packages/rdiff_backup/librsync.py:153: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
  self.maker = _librsync.new_deltamaker(sig_string)
/home/felix/projects/rdiff-backup/.tox/py38/lib/python3.8/site-packages/rdiff_backup/librsync.py:90: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
  self.eof, len_inbuf_read, cycle_out = self.maker.cycle(self.inbuf)
../home/felix/projects/rdiff-backup/.tox/py38/lib/python3.8/site-packages/rdiff_backup/librsync.py:90: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
  self.eof, len_inbuf_read, cycle_out = self.maker.cycle(self.inbuf)
F/home/felix/projects/rdiff-backup/.tox/py38/lib/python3.8/site-packages/rdiff_backup/librsync.py:90: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
  self.eof, len_inbuf_read, cycle_out = self.maker.cycle(self.inbuf)
F/home/felix/projects/rdiff-backup/.tox/py38/lib/python3.8/site-packages/rdiff_backup/librsync.py:218: DeprecationWarning: PY_SSIZE_T_CLEAN will be required for '#' formats
  eof, len_buf_read, cycle_out = self.sig_maker.cycle(self.buffer)
.
======================================================================
FAIL: testSigFile (__main__.LibrsyncTest)
Make sure SigFile generates same data as rdiff, blocksize 512
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testing/librsynctest.py", line 62, in testSigFile
    self.sig_file_test_helper(512, 5)
  File "testing/librsynctest.py", line 51, in sig_file_test_helper
    assert rdiff_sig == librsync_sig, \
AssertionError: (564, 564)

======================================================================
FAIL: testSigFile2 (__main__.LibrsyncTest)
Test SigFile like above, but try various blocksize
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testing/librsynctest.py", line 66, in testSigFile2
    self.sig_file_test_helper(2048, 1, 60000)
  File "testing/librsynctest.py", line 51, in sig_file_test_helper
    assert rdiff_sig == librsync_sig, \
AssertionError: (372, 372)

----------------------------------------------------------------------
Ran 5 tests in 0.119s

FAILED (failures=2)
ERROR: InvocationError for command /home/felix/projects/rdiff-backup/.tox/py38/bin/python testing/librsynctest.py (exited with code 1)
________________________________________________________________ summary ________________________________________________________________
ERROR:   py38: commands failed
```

In distro packaging we should not have internet access or root access during the build (docker is of course even more impossible). I worked around the tarball extraction by using fakeroot but of course some mknod'd nodes are missing, so I also ignore the related test failures.

--- a/testing/librsynctest.py
+++ b/testing/librsynctest.py
@@ -33,7 +33,12 @@ class LibrsyncTest(unittest.TestCase):
         for i in range(iterations):
             MakeRandomFile(self.basis.path, file_len)
             self._clean_file(self.sig)
-            if b'-H' in subprocess.check_output(["rdiff", "--help"]):
+            rdiff_help_text = subprocess.check_output(["rdiff", "--help"])
+            if b'-R' in rdiff_help_text:
+                assert not os.system(
+                    b"rdiff -b %i -R rollup -H md4 signature %b %b" %
+                    (blocksize, self.basis.path, self.sig.path))
+            elif b'-H' in rdiff_help_text:
                 assert not os.system(
                     b"rdiff -b %i -H md4 signature %b %b" %
                     (blocksize, self.basis.path, self.sig.path))

